### ISSUE TITLE
SBGN-ML interoperability: Glyph IDs must not contain numeric as first digit

### ISSUE THREAD

**Issue by lknegendorf** [iVis-at-Bilkent/newt#679]
Hi,

we created a SBGN PD map in Newt and wanted to validate the exchangeability of the map in other tools supporting SBGN. We exported it as SBGN-ML version 0.2 and tried to import the SBGN-ML into VANTED (v2.8) with the corresponding SGBN-ED plugin (v1.6). This attempt throws a cvc-datatype-valid.1.2.1 exception.
We recognized, that glyph IDs with numerical value as first digit are not compatible with this tool. A manual renaming of all such glyph IDs in the SBGN-ML solved the issue for us.

Just wanted to let you know.
Cheers!

**Comment by ugurdogrusoz** [comment]
@lknegendorf Thanks for the feedback. If you create a map in Newt, the IDs for the glyphs and arcs should be starting with "nwtN_" for glyphs and "nwtE_" for arcs such as "nwtN_46cfbb24-c95b-4816-9e8a-2ae2c39bab1f".
Can you perhaps tell us how to reproduce the map you describe or send us the SBGML file for investigation?

**Comment by lknegendorf** [comment]
@ugurdogrusoz Thanks for your reply.
Sure, the map we created was this one: https://github.com/ahodelin/Bachmann_Archive/blob/5cf92db95f79aae0db91d278377c47d4b3628a82/model/sbgn/BachmannPD_source_newt.nwt
We exported the map as SBGN-ML 0.2 and could observe that not all glyph IDs begin with "nwtN_" or "nwtE", but e.g. just "2c837646-ac34-039d-c363-482176c80d4a" or "ea850968-a714-4d00-f035-28c3e687ea0e".
We then had to rename all glyph IDs starting with a numeric.
This is the SBGN-ML file after manual renaming: https://github.com/ahodelin/Bachmann_Archive/blob/5cf92db95f79aae0db91d278377c47d4b3628a82/model/sbgn/BachmannPD_newt.sbgn
You could see there, that there are still glyph IDs without the prefix, e.g. in [line 23](https://github.com/ahodelin/Bachmann_Archive/blob/5cf92db95f79aae0db91d278377c47d4b3628a82/model/sbgn/BachmannPD_newt.sbgn#L23)  "a4654a8c-1e3c-ae08-59c3-60671c6d080e".

**Comment by ugurdogrusoz** [comment]
Interesting! Were the maps with this problem constructed *completely* manually using Newt? Or were they perhaps partially taken from somewhere else and modified in Newt?

**Comment by ugurdogrusoz** [comment]
If it was completely constructed in Newt, when was it first started? Looks like we changed the IDing mechanism (start with "nwt") a couple of years ago and it shouldn't be possible to create glyphs with such IDs any longer.

**Comment by lknegendorf** [comment]
Yes, the maps were constructed completely manually in Newt, we started the construction in June 2021.
For me, it seems the problem is only with macromolecules, nucleic acid features and complexes (only some). We did not observe glyph IDs for processes or arcs, which had to be renamed.

**Comment by ugurdogrusoz** [comment]
That's really odd. When you save your maps, do you use "File > Export > SBGNML Plain" or just "File > Save.." (.nwt)?

**Comment by lknegendorf** [comment]
We used both, but the problem I describe happends just when exporting it via "File > Export > SBGNML Plain" and then trying to reimport it into another tool (we used Vanted/SBGN-ED).

**Comment by ugurdogrusoz** [comment]
@hasanbalci identified the cause of this (when cloning graph objects we neglect to prepend "newt_N/E", will fix it sometime.

**Comment by hasanbalci** [comment]
@sinanhavus Let's add the new options you introduced in clipboard extension [here](https://github.com/iVis-at-Bilkent/newt/blob/18cd119938cdc4bf5cb08c07d1b41ec930cfc30c/app/js/app-cy.js#L401) and see if it works.
